types/src/v2/request.rs

```diff
@@ -21,22 +21,24 @@ pub struct JsonRpcRequest<'a> {
 }
 
 /// Invalid request with known request ID.
-#[derive(Deserialize, Debug)]
+#[derive(Deserialize, Debug, PartialEq)]
 pub struct JsonRpcInvalidRequest<'a> {
 	/// Request ID
 	#[serde(borrow)]
-	pub id: Option<&'a RawValue>,
+	pub id: Id<'a>,
 }
 
 /// JSON-RPC notification (a request object without a request ID).
 #[derive(Serialize, Deserialize, Debug)]
+#[serde(deny_unknown_fields)]
 pub struct JsonRpcNotification<'a> {
 	/// JSON-RPC version.
 	pub jsonrpc: TwoPointZero,
 	/// Name of the method to be invoked.
 	pub method: &'a str,
 	/// Parameter values of the request.
-	pub params: JsonRpcNotificationParams<'a>,
+	#[serde(borrow)]
+	pub params: Option<&'a RawValue>,
```

Comment:
In another PR, hopefully merge JsonRpcParams/RpcParams and replace RawValue here,

That's the reason why some de-serialize tests still fail